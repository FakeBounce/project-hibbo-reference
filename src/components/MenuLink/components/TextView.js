import React from 'react';
import { Text, Animated, View } from 'react-native';
import {
  oneOfType,
  string,
  number,
  instanceOf,
  object,
  array,
} from 'prop-types';
import EStyleSheet from 'react-native-extended-stylesheet';

const styles = EStyleSheet.create({
  text: {
    color: '$colors.darkGrey',
    fontSize: 36,
    paddingTop: 7,
    fontFamily: '$fonts.circularBook',
  },
  textIpad: {
    color: '$colors.darkGrey',
    fontSize: 46,
    lineHeight: 100,
    paddingTop: 0,
    fontFamily: '$fonts.circularBook',
  },
  textHeight: {
    height: 62,
  },
  textHeightIpad: {
    height: 100,
  },
});

const TextView = ({ text, style, device, linkScale, linkOpacity }) => {
  return (
    <View>
      {device === 'iphone' && (
        <Animated.View
          style={[
            styles.textHeight,
            {
              opacity: linkOpacity,
              transform: [{ scale: linkScale }],
            },
          ]}
        >
          <Text style={[styles.text, style]}>{text}</Text>
        </Animated.View>
      )}
      {device === 'ipad' && (
        <Animated.View
          style={[
            styles.textHeightIpad,
            {
              opacity: linkOpacity,
              transform: [{ scale: linkScale }],
            },
          ]}
        >
          <Text style={[styles.textIpad, style]}>{text}</Text>
        </Animated.View>
      )}
    </View>
  );
};

TextView.defaultProps = {
  style: {},
  device: 'iphone',
};

TextView.propTypes = {
  text: string.isRequired,
  style: oneOfType([object, number, array]),
  device: string,
  linkScale: instanceOf(Animated.Value).isRequired,
  linkOpacity: instanceOf(Animated.Value).isRequired,
};

export default TextView;
