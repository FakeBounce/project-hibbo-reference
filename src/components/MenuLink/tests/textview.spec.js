import { Animated } from 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import TextView from '../components/TextView';

describe('TextView', () => {
  it('should render the component for iphone', () => {
    const text = 'test';
    const linkScale = new Animated.Value(1);
    const linkOpacity = new Animated.Value(1);
    const device = 'iphone';

    const textView = renderer
      .create(
        <TextView text={text} linkScale={linkScale} linkOpacity={linkOpacity} device={device} />,
      )
      .toJSON();

    expect(textView).toMatchSnapshot();
  });

  it('should render the component for ipad', () => {
    const text = 'test';
    const linkScale = new Animated.Value(1);
    const linkOpacity = new Animated.Value(1);
    const device = 'ipad';

    const textView = renderer
      .create(
        <TextView text={text} linkScale={linkScale} linkOpacity={linkOpacity} device={device} />,
      )
      .toJSON();

    expect(textView).toMatchSnapshot();
  });
});
