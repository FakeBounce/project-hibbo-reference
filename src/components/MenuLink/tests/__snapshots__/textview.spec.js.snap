// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextView should render the component for ipad 1`] = `
<View>
  <View
    collapsable={undefined}
    style={
      Object {
        "height": 100,
        "opacity": 1,
        "transform": Array [
          Object {
            "scale": 1,
          },
        ],
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          undefined,
          Object {},
        ]
      }
    >
      test
    </Text>
  </View>
</View>
`;

exports[`TextView should render the component for iphone 1`] = `
<View>
  <View
    collapsable={undefined}
    style={
      Object {
        "height": 62,
        "opacity": 1,
        "transform": Array [
          Object {
            "scale": 1,
          },
        ],
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          undefined,
          Object {},
        ]
      }
    >
      test
    </Text>
  </View>
</View>
`;
